core/src/surplus.ts

```diff
@@ -86,3 +87,48 @@ export const fetchActiveSurplusAuctions = async function (network: string): Prom
     }
     return surplusAuctions;
 };
+const surplusAuctionRestartedCallback = async function (
+    network: string,
+    auctionIndex: number
+): Promise<SurplusAuction> {
+    const restartedAuction = await getActiveSurplusAuctionOrUndefined(network, auctionIndex);
+    if (!restartedAuction) {
+        throw new Error('Failed to refetch the restarted auction.');
+    }
+    return restartedAuction;
+};
+
+export const restartSurplusAuction = async function (
+    network: string,
+    auctionIndex: number,
+    notifier?: Notifier
+): Promise<SurplusAuction> {
+    await executeTransaction(network, 'MCD_FLAP', 'tick', [auctionIndex], notifier);
+    return await surplusAuctionRestartedCallback(network, auctionIndex);
```

Comment:
I don't think that refetching should be part of the `restart` function (and you also don't use this functionality in the store) so please remove it here as well as `surplusAuctionRestartedCallback`